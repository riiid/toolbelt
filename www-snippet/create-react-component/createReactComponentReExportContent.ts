import { CreateReactComponentOption } from "./CreateReactComponentOption.ts";

export const createReactComponentReExportContent = (
  name: string,
  option: CreateReactComponentOption,
) => {
  return option.onlyNamedExport
    ? `export * from "./${name}";
  `
    : `export { default } from "./${name}";
export * from "./${name}";
`;
};
