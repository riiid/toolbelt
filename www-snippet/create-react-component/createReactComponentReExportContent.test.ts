import { createReactComponentReExportContent } from "./createReactComponentReExportContent.ts";
import { assertSnapshot } from "https://deno.land/std@0.138.0/testing/snapshot.ts";

Deno.test(
  `should return correct react component re-export.`,
  async (t) => {
    await assertSnapshot(
      t,
      createReactComponentReExportContent("LoremIpsum", {}),
    );
    await assertSnapshot(
      t,
      createReactComponentReExportContent("LoremIpsum", {
        onlyNamedExport: true,
      }),
    );
  },
);
