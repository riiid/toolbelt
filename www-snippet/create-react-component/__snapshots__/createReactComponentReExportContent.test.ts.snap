export const snapshot = {};

snapshot[`should return correct react component re-export. 1`] = `
'export { default } from "./LoremIpsum";
export * from "./LoremIpsum";
'
`;

snapshot[`should return correct react component re-export. 2`] = `
'export * from "./LoremIpsum";
  '
`;
